tests/referral_tokens/test_exchange_referral_token.py

```diff
@@ -0,0 +1,179 @@
+import pytest
+from graphql_relay import to_global_id
+from creator.studies.models import Study
+from django.contrib.auth.models import Group
+from creator.referral_tokens.models import ReferralToken
+from creator.studies.factories import StudyFactory
+from creator.referral_tokens.factories import ReferralTokenFactory
+
+EXCHANGE_REFERRALTOKEN = """
+mutation ($token: ID!) {
+  exchangeReferralToken(token: $token) {
+    message
+    user {
+      email
+      studies {
+        edges {
+          node {
+            id
+          }
+        }
+      }
+      groups {
+        edges {
+          node {
+            id
+          }
+        }
+      }
+    }
+  }
+}
+"""
+
+CREATE_REFERRALTOKEN = """
+mutation newReferralToken($input: ReferralTokenInput!) {
+  createReferralToken(input: $input) {
+    referralToken {
+      uuid
+    }
+  }
+}
+"""
+
+
+@pytest.mark.parametrize(
+    "user_group,allowed",
+    [
+        ("Administrators", True),
+        ("Services", True),
+        ("Developers", True),
+        ("Investigators", True),
+        ("Bioinformatics", True),
+        (None, False),
+    ],
+)
+def test_exchange_referral_token(db, clients, user_group, allowed):
+    """
+    All login user can exchange referral token
+    """
+    client = clients.get(user_group)
+
+    referral_token = ReferralTokenFactory()
+    assert ReferralToken.objects.first().claimed is False
+
+    resp = client.post(
+        "/graphql",
+        data={
+            "query": EXCHANGE_REFERRALTOKEN,
+            "variables": {
+                "token": to_global_id("ReferralTokenNode", referral_token.uuid)
+            },
+        },
+        content_type="application/json",
+    )
+
+    if allowed:
+        assert (
+            resp.json()["data"]["exchangeReferralToken"]["message"]
+            == "Studies are added to this user"
+        )
+        assert ReferralToken.objects.first().claimed is True
+    else:
+        assert resp.json()["errors"][0]["message"] == "Not allowed"
+
+
+def test_exchange_referral_token_invalid(db, clients):
```

Comment:
Would be more descriptive and name it `test_exchange_referral_token_no_multiple_claim`

Reply:
You also want another test here that checks that an expired token cannot be claimed.
You can do that by either creating a token and setting the `created_at` date sometime in the future after expiration,
or by changing the expiration time to a negative number using the `settings` fixture.